src/service/video/docking.js

```diff
@@ -84,13 +91,16 @@ export const RelativeY = {TOP: 0, BOTTOM: 1};
 /** @enum */
 export const Direction = {UP: 1, DOWN: -1};
 
+/** @enum {string} */
+export const Actions = {DOCK: 'dock', UNDOCK: 'undock'};
+
 
 /**
  * @struct @typedef {{
  *  video: !../../video-interface.VideoOrBaseElementDef,
- *  posX: !RelativeX,
- *  posY: !RelativeY,
+ *  target: !DockTargetDef,
  *  step: number,
+ *  triggeredDock: (boolean|undefined),
```

Comment:
Tri-state booleans should usually be avoided. It doesn't look like you explicitly use the undefined state anywhere, so consider removing it and initializing the value to `false`

Reply:
Done.